lib/sisimai/bite/email/kddi.rb

```diff
@@ -0,0 +1,166 @@
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            next if readcursor & Indicators[:deliverystatus] == 0
+            next if e.empty?
+
+            v = dscontents[-1]
+            if cv = e.match(/\A[ \t]+Could not be delivered to: [<]([^ ]+[@][^ ]+)[>]/)
+              # Your mail sent on: Thu, 29 Apr 2010 11:04:47 +0900
+              #     Could not be delivered to: <******@**.***.**>
+              #     As their mailbox is full.
+              if v['recipient']
+                # There are multiple recipient addresses in the message body.
+                push dscontents << Sisimai::Bite.DELIVERYSTATUS
+                v = dscontents[-1]
+              end
+
+              r = Sisimai::Address.s3s4(cv[1])
+              if Sisimai::RFC5322.is_emailaddress(r)
+                v['recipient'] = r
+                recipients += 1
+              end
+
+            elsif cv = e.match(/Your mail sent on: (.+)\z/)
+              # Your mail sent on: Thu, 29 Apr 2010 11:04:47 +0900
+              v['date'] = cv[1]
+
+            else
+              #     As their mailbox is full.
+              v['diagnosis'] ||= ''
+              v['diagnosis']  += e + ' ' if e =~ /\A[ \t]+/
+            end
+          end
+        end
+
+        return nil if recipients.zero?
+        require 'sisimai/smtp/status'
+
+        dscontents.map do |e|
+          e['agent']     = self.smtpagent
+          e['diagnosis'] = Sisimai::String.sweep(e['diagnosis'])
+
+          if mhead['x-spasign'] && mhead['x-spasign'] == 'NG'
+            # Content-Type: text/plain; ..., X-SPASIGN: NG (spamghetti, au by KDDI)
+            # Filtered recipient returns message that include 'X-SPASIGN' header
+            e['reason'] = 'filtered'
+
+          else
+            if e['command'] == 'RCPT'
+              # set "userunknown" when the remote server rejected after RCPT
+              # command.
+              e['reason'] = 'userunknown'
+            else
+              # SMTP command is not RCPT
+              ReFailure.each_key do |r|
+                # Verify each regular expression of session errors
+                next unless e['diagnosis'] =~ ReFailure[r]
+                e['reason'] = r.to_s
+                break
+              end
```

Comment:
__[rubocop]__
* Style/IfInsideElse: Convert `if` nested inside `else` to `elsif`.

<!-- Comment by [SideCI](https://sideci.com) -->